cms/djangoapps/course_to_library_import/helpers.py

```diff
@@ -187,38 +191,191 @@ def _process_staged_content_files(
             )
 
 
-def flat_import_children(block_to_import, library_key, user_id, staged_content, override):
+def _update_container_components(container_version, component_versions, user_id):
+    return authoring_api.create_next_container_version(
+        container_pk=container_version.container.pk,
+        title=container_version.title,
+        publishable_entities_pks=[
+            cv.container.pk if isinstance(cv, ContainerVersion) else cv.component.pk for cv in component_versions
+        ],
+        entity_version_pks=[cv.pk for cv in component_versions],
+        created=datetime.now(tz=timezone.utc),
+        created_by=user_id,
+        container_version_cls=container_version.__class__,
+    )
+
+
+def import_children(block_to_import, library_key, user_id, staged_content, composition_level, override):
     """
     Import children of a block from staged content into a library.
+    Creates appropriate container hierarchy based on composition_level.
     """
     staged_keys = [UsageKey.from_string(key) for key in staged_content.tags.keys()]
     block_id_to_usage_key = {key.block_id: key for key in staged_keys}
 
-    for child in block_to_import.getchildren():
+    result = []
+
+    def _process_xblock(child):
+        usage_key_str = child.get('url_name')
+
+        if usage_key_str:
+            usage_key = block_id_to_usage_key.get(usage_key_str)
+
+            if usage_key and usage_key in staged_keys:
+                library = ContentLibrary.objects.filter(
+                    org__short_name=library_key.org, slug=library_key.slug
+                ).first()
+                if not library:
+                    raise ValueError(f"Library {library_key} does not exist.")
+
+                return create_block_in_library(
+                    child,
+                    usage_key,
+                    library_key,
+                    user_id,
+                    staged_content.id,
+                    override,
+                )
+
+    block_children = block_to_import.getchildren()
+    if len(block_children) == 0 and composition_level == 'xblock':
+        component_version = _process_xblock(block_to_import)
+        if component_version:
+            return [component_version]
+
+    for child in block_children:
         if child.tag in ('chapter', 'sequential', 'vertical'):
-            flat_import_children(child, library_key, user_id, staged_content, override)
+            child_url_name = child.get('url_name')
+            child_display_name = child.get('display_name', '')
+
+            container_version = create_container(
+                child.tag,
+                child_url_name,
```

Comment:
As I see it, `child_url_name` is used only once, so I suggest:
```suggestion
                key=child.get('url_name'),
```

Reply:
updated